### PR TITLE
Treat Code Analysis warnings as errors in CLI project

### PR DESCRIPTION
**Pull request by bastbu**
# PR for issue #898

## What is being addressed

With this PR we fix existing CA/IDE warnings for the CLI project and treat CA/IDE warnings as errors.

## How is this addressed

Warnings that have many occurrences are addressed on a warning-per-commit basis, while warnings that only have one or two occurrences per CA/IDE warning are addressed in bulk.

### CODE REVIEW

File: Tools/Cli-LoRa-Device-Provisioning/Helpers/IoTDeviceHelper.cs [hunk 1]
```diff
@@ -1353,5 +1332,29 @@ public async Task<bool> RemoveDevice(string devEui, ConfigurationHelper configur
             Console.WriteLine("done.");
             return true;
         }
+
+        private static async Task<bool> ExecuteWithIotHubErrorHandlingAsync(Func<Task> executeAsync, string errorMessageContext = null)
```
Comment (spygi) [inline Tools/Cli-LoRa-Device-Provisioning/Helpers/IoTDeviceHelper.cs:1333]: [nit] do you think this makes sense to keep as a separate method, given that this is referenced only once?

**Review by spygi — APPROVED** [review]
LGTM! Thanks for the atomic commits, made it easier to review this.

File: Tools/Cli-LoRa-Device-Provisioning/Cli-LoRa-Device-Provisioning.csproj [hunk 2]
```diff
@@ -5,9 +5,6 @@
     <TargetFramework>$(TargetFramework)</TargetFramework>
     <RootNamespace>LoRaWan.Tools.CLI</RootNamespace>
     <AssemblyName>loradeviceprovisioning</AssemblyName>
-    <CodeAnalysisTreatWarningsAsErrors>false</CodeAnalysisTreatWarningsAsErrors>
-    <TreatWarningsAsErrors>false</TreatWarningsAsErrors>
-    <EnforceCodeStyleInBuild>false</EnforceCodeStyleInBuild>
```
Comment (spygi) [inline Tools/Cli-LoRa-Device-Provisioning/Cli-LoRa-Device-Provisioning.csproj:10]: If not mistaken this [is disabled](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#enforcecodestyleinbuild) by default, don't we want to enable it?

Comment (bastbu) [reply Tools/Cli-LoRa-Device-Provisioning/Cli-LoRa-Device-Provisioning.csproj:10]: This is automatically picked up from the `Directory.Build.props` that we have in the solution.

File: Tools/Cli-LoRa-Device-Provisioning/Helpers/IoTDeviceHelper.cs (hunk 1 above)
Comment (bastbu) [reply Tools/Cli-LoRa-Device-Provisioning/Helpers/IoTDeviceHelper.cs]: Done in 77aeb36e4dc075aec50b2f18c5c7b254f086a76f

**Pull request merged**